src/shims/unix/env.rs

```diff
@@ -272,6 +272,61 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         interp_ok(Scalar::from_u32(this.get_current_tid()))
     }
 
+    fn uname(&mut self, uname: &OpTy<'tcx>) -> InterpResult<'tcx, Scalar> {
+        let this = self.eval_context_mut();
+        this.assert_target_os_is_unix("uname");
+
+        let uname_ptr = this.read_pointer(uname)?;
+        if this.ptr_is_null(uname_ptr)? {
+            return this.set_last_error_and_return_i32(LibcError("EFAULT"));
+        }
+
+        let mut uname_buf = unsafe { std::mem::zeroed() };
```

Comment:
FYI, we try to avoid unsafe in the interpreter except where it is absolutely necessary.

Reply:
Pushed a commit to initialise all fields to zero. Do note that will make porting to new OS, which might have a different set of fields in `utsname`, a bit more annoying.